### PR TITLE
[FIX]osv: Fix the issue of negative term operator search on many2many fields

### PR DESCRIPTION
**Pull request by ara-odoo**
[BUG FIX] see #4613

### CODE REVIEW

**Comment by KangOl** [comment]
Not the right task

**Comment by KangOl** [comment]
Missings tests

File: openerp/osv/expression.py [hunk 1]
```diff
@@ -164,6 +164,11 @@
 # below, this doesn't necessarily mean that any of those NEGATIVE_TERM_OPERATORS is
 # legal in the processed term.
 NEGATIVE_TERM_OPERATORS = ('!=', 'not like', 'not ilike', 'not in')
+REVERSE_NEGATIVE_OPERATOR = {'!=': '=', 'not like': 'like', 'not ilike': 'ilike', 'not in': 'in'}
+def get_ids_operator(operator):
```
Comment (KangOl) [inline openerp/osv/expression.py]: Please find a better name for this method

**Comment by rim-odoo** [comment]
@ara-odoo Can you put the task link on this PR please?

**Comment by rim-odoo** [comment]
(this is not RDWIP, this is a bugfix)

**Comment by StefanRijnhart** [comment]
How about a proper attribution? This is a port of my fix here https://code.launchpad.net/~therp-nl/openobject-server/7.0-lp1062795-search_m2m_negative_gives_wrong_results/+merge/131632

**Comment by StefanRijnhart** [comment]
And my version contained tests :-P

**Comment by isabellerichard** [comment]
This fix works well on v8, thanks.

**Pull request closed without merging**